src/Connection.php

```diff
@@ -1099,7 +1101,7 @@ public function executeCacheQuery($sql, $params, $types, QueryCacheProfile $qcp)
         }
 
         $connectionParams = $this->params;
-        unset($connectionParams['platform']);
+        unset($connectionParams['platform'], $connectionParams['password'], $connectionParams['url']);
```

Comment:
This looks a lot like the piece of code I previously commented.

Reply:
The `$connectionParams` array is passed to `generateCacheKeys()` in the next line, although we don't need those pieces of information to calculate a good cache key. We have to options here:

* Strip sensitive keys from the `$connectionParams` array or
* flag the second parameter of `generateCacheKeys()` as sensitive as well.

I went for the first option.